test/nodes/test_vectordb_pinecone.py

```
@@ -0,0 +1,342 @@
+
+        assert len(warned_messages) == 0
+
+    def test_missing_collection_name_warns(self, mock_config, warned_messages):
+        """Missing collection name should produce a warning."""
+        config = {'apikey': 'test-key', 'collection': '', 'mode': 'serverless-dense'}
+        ig = self._make_iglobal(config, mock_config)
+        ig.validateConfig()
+        assert any('invalid' in m.lower() or 'missing' in m.lower() for m in warned_messages)
+
+    @pytest.mark.parametrize(
+        'name,violation_fragment',
```

Ruff (0.15.7)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)